phoenicis-repository/src/main/java/org/phoenicis/repository/types/GitRepository.java

```diff
@@ -100,90 +100,73 @@ private void cloneOrUpdate() throws RepositoryException {
+                    .setDirectory(this.localFolder)
+                    .setBranch(this.branch).call()) {
+                LOGGER.info("Finished cloning git-repository '{}' to '{}'", repositoryUri, localFolder);
             } catch (GitAPIException e) {
                 throw new RepositoryException(
-                        String.format("Folder '%s' is no git-repository", this.localFolder.getAbsolutePath()), e);
-            } finally {
-                // close repository to free resources
-                if (gitRepository != null) {
-                    gitRepository.close();
-                }
+                        String.format("Folder '%s' is no git-repository", localFolder.getAbsolutePath()), e);
             }
-        }
-        /*
-         * otherwise open the folder and pull the newest updates from the
-         * repository
-         */
-        else {
+        } else {
+            LOGGER.info("Opening git-repository at '{}'", localFolder);
+
             // if anything doesn't work here, we still have our local checkout
             // e.g. could be that the git repository cannot be accessed, there is not Internet connection etc.
             // TODO: it might make sense to ensure that our local checkout is not empty / a valid git repository
-            try {
-                LOGGER.info("Opening " + this);
-
-                gitRepository = Git.open(localFolder);
-
-                LOGGER.info("Pulling new commits from " + this);
+            try (final Git gitRepository = Git.open(localFolder)) {
+                LOGGER.info("Pulling new commits to '{}'", localFolder);
 
                 gitRepository.pull().call();
+
+                LOGGER.info("Finished pulling new commits to '{}'", localFolder);
             } catch (Exception e) {
                 LOGGER.warn("Could not update {0}. Local checkout will be used.", e);
-            } finally {
-                // close repository to free resources
-                if (gitRepository != null) {
-                    gitRepository.close();
-                }
             }
         }
     }
 
     @Override
     public RepositoryDTO fetchInstallableApplications() {
-        RepositoryDTO result;
         try {
             this.cloneOrUpdateWithLock();
-            result = localRepositoryFactory.createInstance(this.localFolder, this.repositoryUri)
+
+            final RepositoryDTO result = localRepositoryFactory
+                    .createInstance(this.localFolder, this.repositoryUri)
                     .fetchInstallableApplications();
+
+            return result;
         } catch (RepositoryException e) {
             throw new RepositoryException("Could not clone or update git repository", e);
         }
-        return result;
     }
 
     @Override
     public void onDelete() {
         try {
-            FileUtils.deleteDirectory(this.localFolder);
+            FileUtils.deleteDirectory(localFolder);
 
-            LOGGER.info("Deleted " + this);
```

Comment:
Wasn't the purpose of logging `this` to avoid that the repository URL etc. must be printed every time? I thought this was the way how it was implemented before?

Reply:
Yes it was, but in my opinion `this` (or more specific `this.toString()`) returns too much information. Therefore I thought about changing the output to something more specific and self-descriptive

Reply:
Ok.